src/components/chat-room/components/room/room-container.tsx

```diff
@@ -1,40 +1,74 @@
 import * as React from 'react';
+import { Message as ApiMessage, User } from '../../../../api/models';
+import { events } from '../../../../api/socket-events';
+import { SocketContext } from '../../../../socket';
+import { mapMessageToVm } from './mappers';
 import { Room } from './room';
-import { ChatMessage } from './view-model';
-const uuidv4 = require('uuid/v4');
+import { Message } from './view-model';
 
 interface Props {
-  userName: string;
+  user: User | null;
+  onlineUsers: User[];
 }
 
 interface State {
   message: string;
-  conversation: ChatMessage[];
+  conversation: Message[];
 }
 
 export class RoomContainer extends React.Component<Props, State> {
```

Comment:
You can extend `PureComponent` to avoid renders when the user is writting the username in the modal.